src/main/java/org/springframework/hateoas/UriTemplate.java

```diff
@@ -299,4 +310,17 @@ private static void appendToBuilder(UriComponentsBuilder builder, TemplateVariab
 				break;
 		}
 	}
+
+	private static void appendComposite(UriComponentsBuilder builder, String name, Object value) {
+		if (value instanceof List) {
```

Comment:
I would use `Iterable` instead of `List` since that is the type supported by the Java 5 for loop.